src/odb/src/codeGenerator/gen.py

```diff
@@ -264,6 +264,10 @@ def get_json_files(directory):
         else:
             field["setterArgumentType"] = field["getterReturnType"] = field["type"]
 
+        # For fields that we need to free/destroy in the destructor
+        if field["name"] == '_name' and 'no-destruct' not in field["flags"] or "table" in field:
+          klass["has_destructible_fields"] = True
```

Comment:
has_destructible_fields isn't a clear name to me.  All the fields are destructible, what varies is whether custom destructor code is needed.   Also a user might need to write dtor code that isn't captured by the above conditions in the "//User Code" section.

Perhaps has_non_default_destructor ?

Reply:
I think that makes sense, but a better name  may be "needs_non_default_destructor" because klass do not have a destructor yet, wdyt?

Reply:
Works for me

Reply:
Changed!